### ISSUE TITLE
FPP Fails and wont restart after pi cape Rpi28D+ EEPROM license upgrade to 4 ports and xlights controller upload

### ISSUE THREAD

**Issue by SarahDods** [FalconChristmas/fpp#1538]
FPP version 6.2 running on a Pi Zero W with RPi28D+ cape

**Describe the bug and Steps to reproduce**: 
A clear and concise description of what the bug is and the steps to be taken to reproduce. 
1) Purchased and uploaded 2 x 2FPP channel EEPROM licenses to activate 4 port functionality on cape.  Confirmed could see 4 ports in Channel output screen
2) Uploaded xlights layout from Xlights Controller screen button to pi.  No errors (4938 channels total using 3 ports)
3) After upload, FPP end had problems - not sure what happened in what order.  After multiple reboots in attempt to test output ports, have
   a) GPIO pin error on Channel Outputs page
   b) crash log in status
   c) FPP stopped, and not startable
   d) selecting restart brings up a successful restart message, but indicator remains at stopped
   e) hitting 'save' on channel outputs page removes all uploaded information from xlights
   d) rebooting fpp doesn't fix it.

Crash log and screenshots attached
![FPP Channel output error after xlights upload](https://github.com/FalconChristmas/fpp/assets/109121211/f9725761-d01f-419d-9869-deb49a57d77d)
![FPP Channel Outputs correct but FPP stopped after xlights upload](https://github.com/FalconChristmas/fpp/assets/109121211/13620331-edb4-45b4-8232-79d8ea8a9cdf)
![FPP Channel outputs screen showing content wiped after saving channel outputs screen](https://github.com/FalconChristmas/fpp/assets/109121211/baacfba7-eeda-4440-8bde-69a518256f3b)
![FPP Display testing failure message](https://github.com/FalconChristmas/fpp/assets/109121211/ecdb59fd-cfe4-417a-bd49-d3b8ec13380b)
![FPP Display testing page unable to get x models](https://github.com/FalconChristmas/fpp/assets/109121211/d8cbf993-4abc-4461-bf95-a8e8dc34409d)
![FPP status page with error](https://github.com/FalconChristmas/fpp/assets/109121211/249b95dc-fd4e-4170-a01c-3de85a7926af)
[fpp-Pi-6.2-2023-07-16_12-36-06.zip](https://github.com/FalconChristmas/fpp/files/12063186/fpp-Pi-6.2-2023-07-16_12-36-06.zip)
![Xlights controller screenshot](https://github.com/FalconChristmas/fpp/assets/109121211/df311c55-6483-446e-9f2c-44739e392a13)
[xLights tree node layout.pdf](https://github.com/FalconChristmas/fpp/files/12063187/xLights.tree.node.layout.pdf)
![Xlights tree setup screenshot](https://github.com/FalconChristmas/fpp/assets/109121211/2daf5f57-5800-442f-9fac-aa08571cb5d7)

**Comment by xelfer** [comment]
Running into this exact issue now. Syslog on the system throwing segfaults:

```
Aug 14 10:22:19 Jacket fppd[3145]: fppd: malloc.c:4036: _int_malloc: Assertion `(unsigned long) (size) >= (unsigned long) (nb)' failed.
Aug 14 10:22:19 Jacket fppd[3145]: Magick: abort due to signal 6 (SIGABRT) "Abort"...
Aug 14 10:22:19 Jacket systemd[1]: fppd.service: Main process exited, code=killed, status=6/ABRT
Aug 14 10:22:21 Jacket systemd[1]: fppd.service: Failed with result 'signal'.
Aug 14 10:22:21 Jacket systemd[1]: Failed to start FPPd.
```

Did you get anywhere with this @SarahDods ?

**Comment by SarahDods** [comment]
Sadly no. I’ve just upgraded a second pi to 4 channels and it’s still there. Workaround is to go back to basic pi eeprom, reapply licenses, then manually copy channel info like we used to.  Painful but gets there. Hopefully they’ll fix it soon. SarahOn 14 Aug 2023, at 10:31 am, Nicolas Triantafillou ***@***.***> wrote:﻿
Running into this exact issue now. Syslog on the system throwing segfaults:
Aug 14 10:22:19 Jacket fppd[3145]: fppd: malloc.c:4036: _int_malloc: Assertion `(unsigned long) (size) >= (unsigned long) (nb)' failed.
Aug 14 10:22:19 Jacket fppd[3145]: Magick: abort due to signal 6 (SIGABRT) "Abort"...
Aug 14 10:22:19 Jacket systemd[1]: fppd.service: Main process exited, code=killed, status=6/ABRT
Aug 14 10:22:21 Jacket systemd[1]: fppd.service: Failed with result 'signal'.
Aug 14 10:22:21 Jacket systemd[1]: Failed to start FPPd.

Did you get anywhere with this @SarahDods ?

**Comment by xelfer** [comment]
Yeah, i've spent a lot of time on it today. It appears to be the 4 port configuration that fails copying over from xlights due to an incorrect implementation of the 4 port 28d+ config. 

If you set it to 2, upload config to xlights then manually change fpp to 4 ports and configure manually (as you suggested) it goes OK.

**Comment by mjunek** [comment]
The issue here is caused by the xcontroller file in xLights using the RPIWS281x driver, rather than the DPIPixels driver. When FPP then tries to load 4 ports against the old driver, it crashes out.

A PR has been created in the xLights repository to resolve this.
